### PR TITLE
add backend protocol to implementers guide

### LINKED ISSUE #2453: Add Backend Protocol Section to implementators guide
Comment: https://github.com/kubernetes-sigs/gateway-api/pull/2444#discussion_r1346610836

### PR DESCRIPTION
**Pull request by joerober**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
Adds section on backend protocol selection to implementors guide explaining that backend protocol selection exists with link to user guide and link to supporting protocols in GEP-1911. 

**Which issue(s) this PR fixes**:

Fixes #2453 

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by dprotaso** [comment]
/assign @youngnick @shaneutt @robscott 

I believe this is the last bit for the backend protocol selection GEP.

File: site-src/guides/implementers.md [hunk 1]
```diff
@@ -355,3 +355,13 @@ or both.
 Implementations that support cross-namespace references MUST watch ReferenceGrant
 and reconcile any ReferenceGrant that points to an object that's referred to by
 an in-scope Gateway API object.
+
+## Backend Protocol
+
+Not all implementations of Gateway API support automatic protocol selection. In some cases protocols are disabled without an explicit opt-in.
```
Comment (robscott) [inline site-src/guides/implementers.md:361]: Nit: It would be good to reword this to focus on the context that this is the Implementer's Guide, and thus intended for implementers to read. What do implementers need to know about this feature? I think the focus in this guide should be to talk about how you might implement it, what this concept usually maps to in common underlying dataplanes like Envoy or NGINX, and where implementers can find more information (I think you already have that part in "References", but it might be good to have more specifics such as the well defined protocols that we recommend implementations support. It would also be good to include keywords such as websockets and HTTP/2 for implementers that may be looking for how to support these concepts with their Gateway API implementation.

**Review by robscott — COMMENTED** [review]
Thanks @joerober!

**Comment by robscott** [comment]
Hey @joerober do you happen to have time to take another look at this one?

**Comment by dprotaso** [comment]
bump

**Review by shaneutt — APPROVED** [review]
/approve

File: site-src/guides/implementers.md (hunk 1 above)
Comment (shaneutt) [reply site-src/guides/implementers.md:361]: @joerober we need to resolve this comment before we proceed, what are your thoughts? :thinking:
